polyglot.js

```diff
@@ -71,29 +104,58 @@ class PolyGlot {
 			actors.push(game.user.character);
 		for (let actor of actors) {
 			try {
-				// Don't duplicate the value in case it's a not an array
-				for (let lang of actor.data.data.traits.languages.value)
-				    this.known_languages.add(lang)
-			} catch (err) { 
+				switch(game.system.id.toLowerCase()){
+					default:
+						// Don't duplicate the value in case it's a not an array
+						for (let lang of actor.data.data.traits.languages.value)
+								this.known_languages.add(lang)
+						break;
+					case "wfrp4e":
+						let re = /Language \((.+)\)/i;
+						for (let item of actor.data.items){
+							let found = item.name.match(re);
+							if(found) {
+								// adding only the descriptive language name, not "Language (XYZ)"
+								this.known_languages.add(found[1]);
+							}
+						}
+						break;
+				}
+			} catch (err) {
 				// Maybe not dnd5e, pf1 or pf2e or corrupted actor data?
 			}
 		}
 		if (this.known_languages.size == 0) {
 			if (game.user.isGM)
 				this.known_languages = new Set(Object.keys(PolyGlot.languages))
-			else
-				this.known_languages.add("common");
+			else {
+				switch(game.system.id.toLowerCase()){
+					default:
+						this.known_languages.add("common");
+						break;
+					case "wfrp4e":
+						this.known_languages.add("Reikspiel");
+				}
+			}
 		}
 		let options = ""
 		for (let lang of this.known_languages) {
 			let label = PolyGlot.languages[lang] || lang
 			options += `<option value="${lang}">${label}</option>`
 		}
-		html.find(".polyglot-lang-select select").html($(options))
+		switch ( game.system.id.toLowerCase() ){
+			default:
+				html.find(".polyglot-lang-select select").html($(options))
+				break;
+			case "wfrp4e":
+				html.find(".polyglot-lang-select-wfrp4e select").html($(options))
```

Comment:
Should not be needed anymore if you don't have a different class just for wfrp4e.
Even if you have no choice but to have a special class for wfrp4e, I would then suggest keeping this select class the same, and just adding an additional one just for wfrp4e, so this case remains the same.